snafu/benchmarks/dnsperf/dnsperf.py

```diff
@@ -0,0 +1,300 @@
+Version {{ dnsperf_version }}
+
+[Status] Command line: {{ command | ORPHRASE }}
+[Status] Sending queries (to {{ ignore }}:{{ ignore }})
+[Status] Started at: {{ start_timestamp | ORPHRASE }}
+[Status] Stopping after {{ time_limit | to_float }} seconds
+</group>
+<group name="data">
+&gt; {{ rcode }} {{ fqdn }} {{ qtype }} {{ rtt_mu_s | to_float }}
+</group>
+<group name="stats">
+[Status] Testing complete (time limit)
+
+Statistics:
+
+  Queries sent:         {{ queries_sent | to_int }}
+  Queries completed:    {{ queries_completed | to_int }} ({{ ignore }}%)
+  Queries lost:         {{ queries_lost | to_int }} ({{ ignore }}%)
+
+  Response codes:       {{ ignore }}
+  Average packet size:  request {{ avg_request_packet_size | to_int }}, response {{ avg_response_packet_size | to_int }} # noqa
+  Run time (s):         {{ duration | to_float }}
+  Queries per second:   {{ throughput | to_float }}
+
+  Average Latency (s):  {{ ORPHRASE }}
+  Latency StdDev (s):   {{ ignore }}
+</group>"""
+
+    args = (
+        ConfigArgument(
+            "-q",
+            "--queries",
+            dest="query_filepath",
+            env_var="QUERIES",
+            required=True,
+            help="filepath to a list of DNS queries",
+        ),
+        ConfigArgument(
+            "-a",
+            "--address",
+            dest="address",
+            help="IPv4 address of DNS server to test",
+            default="172.30.0.10",
+        ),
+        ConfigArgument("-p", "--port", dest="port", help="Port on which DNS packets are sent.", default="53"),
+        ConfigArgument(
+            "-c",
+            "--clients",
+            dest="clients",
+            help="Quantity of clients to act as. (1 client : 1 thread)",
+            default="1",
+        ),
+        ConfigArgument(
+            "-s",
+            "--cache-size",
+            dest="cache_size",
+            env_var="CACHE_SIZE",
+            help="Quantity of entries allowed in cache",
+        ),
+        ConfigArgument(
+            "-l",
+            "--time-limit",
+            dest="time_limit",
+            env_var="TIME_LIMIT",
+            help="Intended duration of test in seconds",
+            default=".05",
+        ),
+        ConfigArgument(
+            "-m",
+            "--transport-mode",
+            dest="transport_mode",
+            help="set transport mode: udp, tcp, dot",
+            default="udp",
+        ),
+        ConfigArgument("--network-policy", dest="networkpolicy", env_var="networkpolicy"),
+        ConfigArgument("--pod-id", dest="pod_id", env_var="my_pod_idx", default=None),
+        ConfigArgument("--steps", dest="steps", default=5),
+    )
+
+    def setup(self) -> bool:
+        """Setup dnsperf benchmark."""
+        self.logger.info("Setting up dnsperf benchmark.")
+        self.config.parse_args()
+
+        if self.config.pod_id:
+            self.tmp_filepath += f"-{self.pod_id}"
+        if check_file(self.tmp_filepath):
```

Comment:
Is this tmp file used for something outside of the benchmark? I included this into the tutorial in the docs as an example for setup/cleanup tasks, but it isn't required.

Reply:
I thought we might want it for a multithreaded version that handles more than one pod. Other than that, I don't think we need it.